src/server/zset_family.cc

```diff
@@ -926,6 +960,68 @@ OpResult<StoreArgs> ParseStoreArgs(CmdArgList args) {
   return store_args;
 };
 
+struct UnionArgs {
+  StoreArgs store_args;
+  bool with_scores = false;
+};
+
+OpResult<UnionArgs> ParseUnionArgs(CmdArgList args) {
+  string_view arg_num_keys = "3";
+  StoreArgs store_args;
+  bool with_scores = false;
+
+  if (!absl::SimpleAtoi(arg_num_keys, &store_args.num_keys)) {
+    return OpStatus::SYNTAX_ERR;
+  }
+  DCHECK_GE(args.size(), 1 + store_args.num_keys);
+
```

Comment:
Maybe add a todo, that as soon as this is not fixed 3-arg, this should be an `if (args < num_keys) return err`